iceoryx_hoofs/test/moduletests/test_cxx_filesystem.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/cxx/filesystem.hpp"
+#include "iceoryx_hoofs/testing/mocks/logger_mock.hpp"
+#include "test.hpp"
+
+namespace
+{
+using namespace ::testing;
+using namespace iox::cxx;
+
+using permsBaseType_t = std::underlying_type<perms>::type;
```

Comment:
or just `underlying_t` for brevity. Or `base_t` to be consistent with the naming in the tests.

could maybe even create a helper function `underlying_t toUnderlying(const perms&);` to shorten the many `static_cast`s in the tests. (or `toBase ...`)